GLOB sdist-make: /code/modules/invenio-oaiserver/setup.py
c1 inst-nodeps: /code/modules/invenio-oaiserver/.tox/.tmp/package/1/invenio-oaiserver-1.0.0.zip
c1 installed: alabaster==0.7.13,alembic==0.9.6,amqp==2.6.0,angular-gettext-babel==0.3,aniso8601==8.0.0,arrow==0.12.1,asn1crypto==0.23.0,atomicwrites==1.4.1,attrs==17.4.0,b2handle==1.1.2,Babel==2.5.1,bagit==1.7.0,beautifulsoup4==4.9.3,bibtexparser==1.0.1,billiard==3.6.3.0,binaryornot==0.4.4,bleach==3.1.0,blinker==1.4,boto3==1.7.84,botocore==1.10.84,cachelib==0.1,cachetools==4.2.4,cchardet==2.1.1,celery==4.4.4,certifi==2017.11.5,cffi==1.11.2,chardet==3.0.4,citeproc-py==0.5.1,citeproc-py-styles==0.1.2,click==6.7,cookiecutter==1.6.0,counter-robots==2018.6,coverage==6.2,cryptography==2.1.4,datacite==1.0.1,DateTime==4.9,decorator==4.1.2,defusedxml==0.5.0,dictdiffer==0.7.0,distlib==0.3.6,dnspython==2.2.1,docutils==0.18.1,dojson==1.3.2,elasticsearch==6.1.1,elasticsearch-dsl==6.4.0,elementpath==1.0.6,email-validator==1.0.5,entrypoints==0.2.3,feedgen==0.7.0,filelock==3.4.1,Flask==1.1.0,Flask-Admin==1.5.3,Flask-Alembic==2.0.1,Flask-Assets==0.12,Flask-BabelEx==0.9.4,Flask-Breadcrumbs==0.5.0,Flask-Caching==1.10.1,Flask-CeleryExt==0.3.4,Flask-Collect==1.2.2,Flask-Cors==3.0.3,Flask-DebugToolbar==0.13.1,Flask-IIIF==0.6.1,Flask-KVSession==0.6.2,Flask-Limiter==1.1.0,Flask-Login==0.4.1,Flask-Mail==0.9.1,flask-marshmallow==0.14.0,Flask-Menu==0.6.0,-e git+https://github.com/RCOSDP/flask-oauthlib.git@d47fafb1558e261f98ca916355346a9ca7b4a68e#egg=Flask_OAuthlib,Flask-Plugins==1.6.1,Flask-Principal==0.4.0,Flask-RESTful==0.3.8,Flask-Security==3.0.0,flask-shell-ipython==0.4.1,Flask-Sitemap==0.4.0,Flask-SQLAlchemy==2.3.2,flask-talisman==0.4.1,Flask-WTF==0.14.3,-e git+https://github.com/RCOSDP/pyfpdf.git@f9b032148283d535cabc7789858081c80de36fef#egg=fpdf,frozendict==2.3.8,fs==0.5.4,ftfy==4.4.3,future==0.16.0,github3.py==1.1.0,html5lib==1.0.1,idna==2.6,iiif-prezi==0.3.0,imagesize==1.4.1,importlib-metadata==4.8.3,importlib-resources==5.4.0,infinity==1.4,iniconfig==1.1.1,intervals==0.8.0,invenio-access==1.1.0,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=invenio_accounts&subdirectory=modules/invenio-accounts,invenio-admin==1.1.2,invenio-app==1.1.0,invenio-assets==1.0.0,invenio-base==1.0.2,invenio-cache==1.0.0,invenio-celery==1.1.3,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=invenio_communities&subdirectory=modules/invenio-communities,invenio-config==1.0.0,invenio-csl-rest==1.0.0a1,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=invenio_db&subdirectory=modules/invenio-db,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=invenio_deposit&subdirectory=modules/invenio-deposit,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=invenio_files_rest&subdirectory=modules/invenio-files-rest,invenio-formatter==1.0.0b3,invenio-i18n==1.0.0,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=invenio_iiif&subdirectory=modules/invenio-iiif,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=invenio_indexer&subdirectory=modules/invenio-indexer,invenio-jsonschemas==1.0.0,invenio-logging==1.0.0b3,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=invenio_mail&subdirectory=modules/invenio-mail,invenio-marc21==1.0.0a8,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=invenio_oaiharvester&subdirectory=modules/invenio-oaiharvester,invenio-oaiserver @ file:///code/modules/invenio-oaiserver/.tox/.tmp/package/1/invenio-oaiserver-1.0.0.zip,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=invenio_oauth2server&subdirectory=modules/invenio-oauth2server,invenio-oauthclient==1.0.0,invenio-pidrelations==1.0.0a4,invenio-pidstore==1.0.0,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=invenio_previewer&subdirectory=modules/invenio-previewer,invenio-query-parser==0.6.0,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=invenio_queues&subdirectory=modules/invenio-queues,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=invenio_records&subdirectory=modules/invenio-records,invenio-records-files==1.0.0a10,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=invenio_records_rest&subdirectory=modules/invenio-records-rest,invenio-records-ui==1.0.0,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=invenio_resourcesyncclient&subdirectory=modules/invenio-resourcesyncclient,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=invenio_resourcesyncserver&subdirectory=modules/invenio-resourcesyncserver,invenio-rest==1.1.2,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=invenio_s3&subdirectory=modules/invenio-s3,-e git+https://github.com/RCOSDP/invenio-search.git@cff9744c5dc651893a9c51672c5b8da9adc21e16#egg=invenio_search,-e git+https://github.com/RCOSDP/invenio-search-ui.git@74bd3b2990ff27b39e01b6b31be9a0b5fda3dd0f#egg=invenio_search_ui,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=invenio_stats&subdirectory=modules/invenio-stats,invenio-theme==1.0.0b4,ipaddress==1.0.19,ipython==6.2.1,ipython-genutils==0.2.0,itsdangerous==0.24,jedi==0.11.0,Jinja2==2.10.1,jinja2-cli==0.6.0,jinja2-time==0.2.0,jmespath==0.10.0,jsmin==2.2.2,jsonpatch==1.21,jsonpath-ng==1.5.2,jsonpointer==1.14,jsonref==0.1,jsonresolver==0.2.1,jsonschema==2.6.0,jupyter-client==5.2.2,jupyter-core==4.4.0,-e git+https://github.com/RCOSDP/kombu.git@f204fdf078d5e94393c86693f545e2d011f620f5#egg=kombu,limits==1.2.1,lxml==4.1.1,Mako==1.0.7,MarkupSafe==1.1.1,marshmallow==2.20.1,marshmallow-sqlalchemy==0.23.1,maxminddb==1.5.2,maxminddb-geolite2==2017.803,mistune==0.8.3,mock==5.0.2,more-itertools==8.10.0,msgpack==0.6.2,nbconvert==5.3.1,nbformat==4.4.0,netaddr==0.8.0,node-semver==0.1.1,numpy==1.16.1,oauthlib==2.1.0,ordereddict==1.1,packaging==21.3,pandocfilters==1.4.2,parso==0.1.0,passlib==1.7.1,pbr==5.11.1,pexpect==4.3.0,pickleshare==0.7.4,Pillow==5.4.1,platformdirs==2.4.0,pluggy==0.13.1,ply==3.11,poyo==0.4.1,prompt-toolkit==1.0.15,psycopg2==2.7.3.2,ptyprocess==0.5.2,py==1.11.0,pycparser==2.18,Pygments==2.2.0,PyJWT==1.5.3,PyLD==2.0.3,pyparsing==3.1.0,-e git+https://github.com/RCOSDP/PyPDF2.git@fefc684a3a74aff6f99e5dff24f9b4dd1c95169d#egg=PyPDF2,pyPEG2==2.15.2,pytest==6.1.2,pytest-cov==4.0.0,pytest-mock==3.6.1,python-dateutil==2.6.1,python-editor==1.0.3,python-geoip==1.2,pytz==2017.3,pyzmq==17.0.0,redis==2.10.6,requests==2.18.4,requests-oauthlib==1.1.0,resync==1.0.9,s3fs==0.1.6,s3transfer==0.1.13,selenium==3.141.0,Sickle==0.6.4,simplegeneric==0.8.1,simplejson==3.12.0,simplekv==0.11.2,six==1.16.0,snowballstemmer==2.2.0,soupsieve==2.3.2.post1,speaklater==1.3,Sphinx==1.8.4,sphinxcontrib-serializinghtml==1.1.5,sphinxcontrib-websupport==1.2.4,SQLAlchemy==1.2.19,SQLAlchemy-Continuum==1.3.6,SQLAlchemy-Utils==0.35.0,stevedore==3.5.2,sword3common==0.1.1,testpath==0.3.1,toml==0.10.2,tomli==1.2.3,tornado==4.5.3,tox==3.28.0,traitlets==4.3.2,typing_extensions==4.1.1,ua-parser==0.7.3,uritemplate==4.1.1,uritools==2.1.0,urllib3==1.22,uWSGI==2.0.21,uwsgitop==0.11,validators==0.12.0,vine==1.3.0,virtualenv==20.17.1,virtualenv-clone==0.5.7,virtualenvwrapper==4.8.4,Wand==0.6.1,wcwidth==0.1.7,webargs==5.5.2,webassets==0.12.1,webencodings==0.5.1,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=weko_accounts&subdirectory=modules/weko-accounts,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=weko_admin&subdirectory=modules/weko-admin,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=weko_authors&subdirectory=modules/weko-authors,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=weko_bulkupdate&subdirectory=modules/weko-bulkupdate,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=weko_deposit&subdirectory=modules/weko-deposit,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=weko_gridlayout&subdirectory=modules/weko-gridlayout,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=weko_groups&subdirectory=modules/weko-groups,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=weko_handle&subdirectory=modules/weko-handle,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=weko_index_tree&subdirectory=modules/weko-index-tree,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=weko_indextree_journal&subdirectory=modules/weko-indextree-journal,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=weko_items_autofill&subdirectory=modules/weko-items-autofill,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=weko_items_ui&subdirectory=modules/weko-items-ui,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=weko_itemtypes_ui&subdirectory=modules/weko-itemtypes-ui,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=weko_logging&subdirectory=modules/weko-logging,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=weko_records&subdirectory=modules/weko-records,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=weko_records_ui&subdirectory=modules/weko-records-ui,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=weko_redis&subdirectory=modules/weko-redis,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=weko_schema_ui&subdirectory=modules/weko-schema-ui,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=weko_search_ui&subdirectory=modules/weko-search-ui,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=weko_sitemap&subdirectory=modules/weko-sitemap,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=weko_swordserver&subdirectory=modules/weko-swordserver,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=weko_theme&subdirectory=modules/weko-theme,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=weko_user_profiles&subdirectory=modules/weko-user-profiles,-e git+https://mhaya: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@github.com/RCOSDP/weko.git@9681d8d89512eb9df7e6eb70c311342468e65c50#egg=weko_workflow&subdirectory=modules/weko-workflow,Werkzeug==0.15.2,whichcraft==0.4.1,WTForms==2.1,WTForms-Alchemy==0.16.5,WTForms-Components==0.10.3,xmlschema==0.9.30,xmltodict==0.12.0,zipp==3.6.0,zope.interface==5.5.2
c1 run-test-pre: PYTHONHASHSEED='35030701'
c1 run-test: commands[0] | pytest --cov=invenio_oaiserver tests -v -vv -s --cov-branch --cov-report=term --basetemp=/code/modules/invenio-oaiserver/.tox/c1/tmp
============================= test session starts ==============================
platform linux -- Python 3.6.15, pytest-6.1.2, py-1.11.0, pluggy-0.13.1 -- /code/modules/invenio-oaiserver/.tox/c1/bin/python
cachedir: .tox/c1/.pytest_cache
rootdir: /code/modules/invenio-oaiserver
plugins: celery-4.4.4, mock-3.6.1, cov-4.0.0
collecting ... collected 91 items

tests/test_admin.py::test_admin PASSED
tests/test_admin.py::test_OAISetModelView PASSED
tests/test_admin.py::test_IdentifyModelView PASSED
tests/test_api.py::TestOaiIdentify::test_get_all PASSED
tests/test_api.py::TestOaiIdentify::test_get_count PASSED
tests/test_app.py::test_version PASSED
tests/test_app.py::test_init PASSED
tests/test_fetcher.py::test_oaiid_fetcher PASSED
tests/test_invenio_oaiserver.py::test_version PASSED
tests/test_invenio_oaiserver.py::test_init PASSED
tests/test_invenio_oaiserver.py::test_view PASSED
tests/test_invenio_oaiserver.py::test_view_with_xsl PASSED
tests/test_invenio_oaiserver.py::test_alembic SKIPPED
tests/test_invenio_oaiserver.py::test_AppState PASSED
tests/test_minters.py::test_oaiid_minter PASSED
tests/test_models.py::TestOAISet::test_get_set_by_spec PASSED
tests/test_models.py::test_oaiset_attribute_changed PASSED
tests/test_percolator.py::test_search_pattern_change SKIPPED
tests/test_percolator.py::test_populate_oaisets SKIPPED
tests/test_percolator.py::test_oaiset_add_remove_record PASSED
tests/test_percolator.py::test_create_percolator_mapping PASSED
tests/test_percolator.py::test_percolate_query PASSED
tests/test_percolator.py::test_get_percolator_doc_type PASSED
tests/test_percolator.py::test_new_percolator PASSED
tests/test_percolator.py::test_delete_percolator PASSED
tests/test_percolator.py::test_build_cache PASSED
tests/test_percolator.py::test_get_record_sets PASSED
tests/test_provider.py::test_create PASSED
tests/test_query.py::test_query_string_parser PASSED
tests/test_query.py::test_get_affected_records PASSED
tests/test_query.py::test_get_records FAILED
tests/test_query.py::test_get_records ERROR
tests/test_receivers.py::test_OAIServerUpdater PASSED
tests/test_receivers.py::test_after_insert_oai_set PASSED
tests/test_receivers.py::test_after_update_oai_set PASSED
tests/test_receivers.py::test_after_delete_oai_set PASSED
tests/test_response.py::test_getrecord FAILED
tests/test_response.py::test_getrecord_future_item PASSED
tests/test_response.py::test_listidentifiers ERROR
tests/test_response.py::test_listrecords ERROR
tests/test_response.py::test_envelope PASSED
tests/test_response.py::test_identify PASSED
tests/test_response.py::test_resumption_token PASSED
tests/test_response.py::test_listsets PASSED
tests/test_response.py::test_listmetadataformats PASSED
tests/test_response.py::test_header PASSED
tests/test_response.py::test_extract_paths_from_sets PASSED
tests/test_response.py::test_is_pubdate_in_future PASSED
tests/test_response.py::test_is_private_index PASSED
tests/test_response.py::test_is_private_index_by_public_list PASSED
tests/test_response.py::test_get_error_code_msg PASSED
tests/test_response.py::test_create_identifier_index PASSED
tests/test_response.py::test_check_correct_system_props_mapping FAILED
tests/test_response.py::test_combine_record_file_urls FAILED
tests/test_response.py::test_create_files_url PASSED
tests/test_response.py::test_get_identifier PASSED
tests/test_response.py::test_issue34851_listrecords ERROR
tests/test_response.py::test_issue34851_listidentifiers ERROR
tests/test_resumption_token.py::test_schema_from_verb PASSED
tests/test_resumption_token.py::test_serialize PASSED
tests/test_resumption_token.py::test_ResumptionTokenSchema PASSED
tests/test_tasks.py::test_records_commit PASSED
tests/test_tasks.py::test_update_records_sets PASSED
tests/test_tasks.py::test_update_affected_records PASSED
tests/test_utils.py::test_serializer PASSED
tests/test_utils.py::test_dumps_etree PASSED
tests/test_utils.py::test_datetime_to_datestamp PASSED
tests/test_utils.py::test_eprints_description PASSED
tests/test_utils.py::test_handle_license_free PASSED
tests/test_utils.py::test_get_index_state PASSED
tests/test_utils.py::test_is_output_harvest PASSED
tests/test_verbs.py::test_no_verb PASSED
tests/test_verbs.py::test_wrong_verb PASSED
tests/test_verbs.py::test_identify PASSED
tests/test_verbs.py::test_getrecord_fail ERROR
tests/test_verbs.py::test_identify_with_additional_args PASSED
tests/test_verbs.py::test_listmetadataformats PASSED
tests/test_verbs.py::test_listmetadataformats_record ERROR
tests/test_verbs.py::test_listmetadataformats_record_fail PASSED
tests/test_verbs.py::test_listsets PASSED
tests/test_verbs.py::test_fail_missing_metadataPrefix PASSED
tests/test_verbs.py::test_fail_not_exist_metadataPrefix PASSED
tests/test_verbs.py::test_listrecords_fail_missing_metadataPrefix PASSED
tests/test_verbs.py::test_list_sets_with_resumption_token_and_other_args PASSED
tests/test_verbs.py::test_validate_metadata_prefix PASSED
tests/test_verbs.py::test_validate_duplicate_argument PASSED
tests/test_verbs.py::test_DateTime_from_iso_permissive PASSED
tests/test_verbs.py::test_OAIScheme_validate PASSED
tests/test_views_server.py::test_validation_error PASSED
tests/test_views_server.py::test_pid_error PASSED
tests/test_views_server.py::test_resumptiontoken_error PASSED
tests/test_views_server.py::test_dbsession_clean PASSED

==================================== ERRORS ====================================
____________________ ERROR at teardown of test_get_records _____________________

app = <Flask 'testapp'>

    @pytest.yield_fixture()
    def es_app(app):
        with open(join(dirname(__file__),"data/mappings/item-v1.0.0.json"),"r") as f:
        #with open(join(dirname(__file__),"data/v6/records/record-v1.0.0.json"),"r") as f:
            mapping = json.load(f)
        es = Elasticsearch("http://{}:9200".format(app.config["SEARCH_ELASTIC_HOSTS"]))
    
        es.indices.create(
            index=app.config["INDEXER_DEFAULT_INDEX"],
            body=mapping, ignore=[400, 404]
        )
    
        es.indices.put_alias(
            index=app.config["INDEXER_DEFAULT_INDEX"],
            name=app.config["SEARCH_UI_SEARCH_INDEX"],
            ignore=[400, 404],
        )
        search = InvenioSearch(app, client=es)
        #search.register_mappings('items', 'tests.data')
        yield app
    
        es.indices.delete_alias(
            index=app.config["INDEXER_DEFAULT_INDEX"],
            name=app.config["SEARCH_UI_SEARCH_INDEX"],
>           ignore=[400, 404],
        )

tests/conftest.py:301: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.tox/c1/lib/python3.6/site-packages/elasticsearch/client/utils.py:76: in _wrapped
    return func(*args, params=params, **kwargs)
.tox/c1/lib/python3.6/site-packages/elasticsearch/client/indices.py:423: in delete_alias
    '_alias', name), params=params)
.tox/c1/lib/python3.6/site-packages/elasticsearch/transport.py:314: in perform_request
    status, headers_response, data = connection.perform_request(method, url, params, body, headers=headers, ignore=ignore, timeout=timeout)
.tox/c1/lib/python3.6/site-packages/elasticsearch/connection/http_urllib3.py:163: in perform_request
    self._raise_error(response.status, raw_data)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Urllib3HttpConnection: http://elasticsearch:9200>, status_code = 403
raw_data = '{"error":{"root_cause":[{"type":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow...":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"},"status":403}'

    def _raise_error(self, status_code, raw_data):
        """ Locate appropriate exception and raise it. """
        error_message = raw_data
        additional_info = None
        try:
            if raw_data:
                additional_info = json.loads(raw_data)
                error_message = additional_info.get('error', error_message)
                if isinstance(error_message, dict) and 'type' in error_message:
                    error_message = error_message['type']
        except (ValueError, TypeError) as err:
            logger.warning('Undecodable raw error response from server: %s', err)
    
>       raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
E       elasticsearch.exceptions.AuthorizationException: TransportError(403, 'cluster_block_exception', 'blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];')

.tox/c1/lib/python3.6/site-packages/elasticsearch/connection/base.py:125: AuthorizationException
------------------------------ Captured log call -------------------------------
WARNING  elasticsearch:base.py:97 PUT http://elasticsearch:9200/test-weko-item-v1.0.0/item-v1.0.0/2fb47f17-1422-4fe4-b9d8-1395cc6e1070?version=1&version_type=external_gte [status:403 request:0.004s]
---------------------------- Captured log teardown -----------------------------
WARNING  elasticsearch:base.py:97 DELETE http://elasticsearch:9200/test-weko-item-v1.0.0/_alias/test-weko [status:403 request:0.004s]
____________________ ERROR at setup of test_listidentifiers ____________________

app = <Flask 'testapp'>

    @pytest.yield_fixture()
    def es_app(app):
        with open(join(dirname(__file__),"data/mappings/item-v1.0.0.json"),"r") as f:
        #with open(join(dirname(__file__),"data/v6/records/record-v1.0.0.json"),"r") as f:
            mapping = json.load(f)
        es = Elasticsearch("http://{}:9200".format(app.config["SEARCH_ELASTIC_HOSTS"]))
    
        es.indices.create(
            index=app.config["INDEXER_DEFAULT_INDEX"],
>           body=mapping, ignore=[400, 404]
        )

tests/conftest.py:286: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.tox/c1/lib/python3.6/site-packages/elasticsearch/client/utils.py:76: in _wrapped
    return func(*args, params=params, **kwargs)
.tox/c1/lib/python3.6/site-packages/elasticsearch/client/indices.py:91: in create
    params=params, body=body)
.tox/c1/lib/python3.6/site-packages/elasticsearch/transport.py:314: in perform_request
    status, headers_response, data = connection.perform_request(method, url, params, body, headers=headers, ignore=ignore, timeout=timeout)
.tox/c1/lib/python3.6/site-packages/elasticsearch/connection/http_urllib3.py:163: in perform_request
    self._raise_error(response.status, raw_data)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Urllib3HttpConnection: http://elasticsearch:9200>, status_code = 403
raw_data = '{"error":{"root_cause":[{"type":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow...":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"},"status":403}'

    def _raise_error(self, status_code, raw_data):
        """ Locate appropriate exception and raise it. """
        error_message = raw_data
        additional_info = None
        try:
            if raw_data:
                additional_info = json.loads(raw_data)
                error_message = additional_info.get('error', error_message)
                if isinstance(error_message, dict) and 'type' in error_message:
                    error_message = error_message['type']
        except (ValueError, TypeError) as err:
            logger.warning('Undecodable raw error response from server: %s', err)
    
>       raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
E       elasticsearch.exceptions.AuthorizationException: TransportError(403, 'cluster_block_exception', 'blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];')

.tox/c1/lib/python3.6/site-packages/elasticsearch/connection/base.py:125: AuthorizationException
------------------------------ Captured log setup ------------------------------
WARNING  elasticsearch:base.py:97 PUT http://elasticsearch:9200/test-weko-item-v1.0.0 [status:403 request:0.007s]
______________________ ERROR at setup of test_listrecords ______________________

app = <Flask 'testapp'>

    @pytest.yield_fixture()
    def es_app(app):
        with open(join(dirname(__file__),"data/mappings/item-v1.0.0.json"),"r") as f:
        #with open(join(dirname(__file__),"data/v6/records/record-v1.0.0.json"),"r") as f:
            mapping = json.load(f)
        es = Elasticsearch("http://{}:9200".format(app.config["SEARCH_ELASTIC_HOSTS"]))
    
        es.indices.create(
            index=app.config["INDEXER_DEFAULT_INDEX"],
>           body=mapping, ignore=[400, 404]
        )

tests/conftest.py:286: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.tox/c1/lib/python3.6/site-packages/elasticsearch/client/utils.py:76: in _wrapped
    return func(*args, params=params, **kwargs)
.tox/c1/lib/python3.6/site-packages/elasticsearch/client/indices.py:91: in create
    params=params, body=body)
.tox/c1/lib/python3.6/site-packages/elasticsearch/transport.py:314: in perform_request
    status, headers_response, data = connection.perform_request(method, url, params, body, headers=headers, ignore=ignore, timeout=timeout)
.tox/c1/lib/python3.6/site-packages/elasticsearch/connection/http_urllib3.py:163: in perform_request
    self._raise_error(response.status, raw_data)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Urllib3HttpConnection: http://elasticsearch:9200>, status_code = 403
raw_data = '{"error":{"root_cause":[{"type":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow...":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"},"status":403}'

    def _raise_error(self, status_code, raw_data):
        """ Locate appropriate exception and raise it. """
        error_message = raw_data
        additional_info = None
        try:
            if raw_data:
                additional_info = json.loads(raw_data)
                error_message = additional_info.get('error', error_message)
                if isinstance(error_message, dict) and 'type' in error_message:
                    error_message = error_message['type']
        except (ValueError, TypeError) as err:
            logger.warning('Undecodable raw error response from server: %s', err)
    
>       raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
E       elasticsearch.exceptions.AuthorizationException: TransportError(403, 'cluster_block_exception', 'blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];')

.tox/c1/lib/python3.6/site-packages/elasticsearch/connection/base.py:125: AuthorizationException
------------------------------ Captured log setup ------------------------------
WARNING  elasticsearch:base.py:97 PUT http://elasticsearch:9200/test-weko-item-v1.0.0 [status:403 request:0.006s]
________________ ERROR at setup of test_issue34851_listrecords _________________

app = <Flask 'testapp'>

    @pytest.yield_fixture()
    def es_app(app):
        with open(join(dirname(__file__),"data/mappings/item-v1.0.0.json"),"r") as f:
        #with open(join(dirname(__file__),"data/v6/records/record-v1.0.0.json"),"r") as f:
            mapping = json.load(f)
        es = Elasticsearch("http://{}:9200".format(app.config["SEARCH_ELASTIC_HOSTS"]))
    
        es.indices.create(
            index=app.config["INDEXER_DEFAULT_INDEX"],
>           body=mapping, ignore=[400, 404]
        )

tests/conftest.py:286: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.tox/c1/lib/python3.6/site-packages/elasticsearch/client/utils.py:76: in _wrapped
    return func(*args, params=params, **kwargs)
.tox/c1/lib/python3.6/site-packages/elasticsearch/client/indices.py:91: in create
    params=params, body=body)
.tox/c1/lib/python3.6/site-packages/elasticsearch/transport.py:314: in perform_request
    status, headers_response, data = connection.perform_request(method, url, params, body, headers=headers, ignore=ignore, timeout=timeout)
.tox/c1/lib/python3.6/site-packages/elasticsearch/connection/http_urllib3.py:163: in perform_request
    self._raise_error(response.status, raw_data)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Urllib3HttpConnection: http://elasticsearch:9200>, status_code = 403
raw_data = '{"error":{"root_cause":[{"type":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow...":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"},"status":403}'

    def _raise_error(self, status_code, raw_data):
        """ Locate appropriate exception and raise it. """
        error_message = raw_data
        additional_info = None
        try:
            if raw_data:
                additional_info = json.loads(raw_data)
                error_message = additional_info.get('error', error_message)
                if isinstance(error_message, dict) and 'type' in error_message:
                    error_message = error_message['type']
        except (ValueError, TypeError) as err:
            logger.warning('Undecodable raw error response from server: %s', err)
    
>       raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
E       elasticsearch.exceptions.AuthorizationException: TransportError(403, 'cluster_block_exception', 'blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];')

.tox/c1/lib/python3.6/site-packages/elasticsearch/connection/base.py:125: AuthorizationException
------------------------------ Captured log setup ------------------------------
WARNING  elasticsearch:base.py:97 PUT http://elasticsearch:9200/test-weko-item-v1.0.0 [status:403 request:0.007s]
______________ ERROR at setup of test_issue34851_listidentifiers _______________

app = <Flask 'testapp'>

    @pytest.yield_fixture()
    def es_app(app):
        with open(join(dirname(__file__),"data/mappings/item-v1.0.0.json"),"r") as f:
        #with open(join(dirname(__file__),"data/v6/records/record-v1.0.0.json"),"r") as f:
            mapping = json.load(f)
        es = Elasticsearch("http://{}:9200".format(app.config["SEARCH_ELASTIC_HOSTS"]))
    
        es.indices.create(
            index=app.config["INDEXER_DEFAULT_INDEX"],
>           body=mapping, ignore=[400, 404]
        )

tests/conftest.py:286: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.tox/c1/lib/python3.6/site-packages/elasticsearch/client/utils.py:76: in _wrapped
    return func(*args, params=params, **kwargs)
.tox/c1/lib/python3.6/site-packages/elasticsearch/client/indices.py:91: in create
    params=params, body=body)
.tox/c1/lib/python3.6/site-packages/elasticsearch/transport.py:314: in perform_request
    status, headers_response, data = connection.perform_request(method, url, params, body, headers=headers, ignore=ignore, timeout=timeout)
.tox/c1/lib/python3.6/site-packages/elasticsearch/connection/http_urllib3.py:163: in perform_request
    self._raise_error(response.status, raw_data)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Urllib3HttpConnection: http://elasticsearch:9200>, status_code = 403
raw_data = '{"error":{"root_cause":[{"type":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow...":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"},"status":403}'

    def _raise_error(self, status_code, raw_data):
        """ Locate appropriate exception and raise it. """
        error_message = raw_data
        additional_info = None
        try:
            if raw_data:
                additional_info = json.loads(raw_data)
                error_message = additional_info.get('error', error_message)
                if isinstance(error_message, dict) and 'type' in error_message:
                    error_message = error_message['type']
        except (ValueError, TypeError) as err:
            logger.warning('Undecodable raw error response from server: %s', err)
    
>       raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
E       elasticsearch.exceptions.AuthorizationException: TransportError(403, 'cluster_block_exception', 'blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];')

.tox/c1/lib/python3.6/site-packages/elasticsearch/connection/base.py:125: AuthorizationException
------------------------------ Captured log setup ------------------------------
WARNING  elasticsearch:base.py:97 PUT http://elasticsearch:9200/test-weko-item-v1.0.0 [status:403 request:0.007s]
____________________ ERROR at setup of test_getrecord_fail _____________________

app = <Flask 'testapp'>

    @pytest.yield_fixture()
    def es_app(app):
        with open(join(dirname(__file__),"data/mappings/item-v1.0.0.json"),"r") as f:
        #with open(join(dirname(__file__),"data/v6/records/record-v1.0.0.json"),"r") as f:
            mapping = json.load(f)
        es = Elasticsearch("http://{}:9200".format(app.config["SEARCH_ELASTIC_HOSTS"]))
    
        es.indices.create(
            index=app.config["INDEXER_DEFAULT_INDEX"],
>           body=mapping, ignore=[400, 404]
        )

tests/conftest.py:286: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.tox/c1/lib/python3.6/site-packages/elasticsearch/client/utils.py:76: in _wrapped
    return func(*args, params=params, **kwargs)
.tox/c1/lib/python3.6/site-packages/elasticsearch/client/indices.py:91: in create
    params=params, body=body)
.tox/c1/lib/python3.6/site-packages/elasticsearch/transport.py:314: in perform_request
    status, headers_response, data = connection.perform_request(method, url, params, body, headers=headers, ignore=ignore, timeout=timeout)
.tox/c1/lib/python3.6/site-packages/elasticsearch/connection/http_urllib3.py:163: in perform_request
    self._raise_error(response.status, raw_data)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Urllib3HttpConnection: http://elasticsearch:9200>, status_code = 403
raw_data = '{"error":{"root_cause":[{"type":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow...":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"},"status":403}'

    def _raise_error(self, status_code, raw_data):
        """ Locate appropriate exception and raise it. """
        error_message = raw_data
        additional_info = None
        try:
            if raw_data:
                additional_info = json.loads(raw_data)
                error_message = additional_info.get('error', error_message)
                if isinstance(error_message, dict) and 'type' in error_message:
                    error_message = error_message['type']
        except (ValueError, TypeError) as err:
            logger.warning('Undecodable raw error response from server: %s', err)
    
>       raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
E       elasticsearch.exceptions.AuthorizationException: TransportError(403, 'cluster_block_exception', 'blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];')

.tox/c1/lib/python3.6/site-packages/elasticsearch/connection/base.py:125: AuthorizationException
------------------------------ Captured log setup ------------------------------
WARNING  elasticsearch:base.py:97 PUT http://elasticsearch:9200/test-weko-item-v1.0.0 [status:403 request:0.008s]
______________ ERROR at setup of test_listmetadataformats_record _______________

app = <Flask 'testapp'>

    @pytest.yield_fixture()
    def es_app(app):
        with open(join(dirname(__file__),"data/mappings/item-v1.0.0.json"),"r") as f:
        #with open(join(dirname(__file__),"data/v6/records/record-v1.0.0.json"),"r") as f:
            mapping = json.load(f)
        es = Elasticsearch("http://{}:9200".format(app.config["SEARCH_ELASTIC_HOSTS"]))
    
        es.indices.create(
            index=app.config["INDEXER_DEFAULT_INDEX"],
>           body=mapping, ignore=[400, 404]
        )

tests/conftest.py:286: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.tox/c1/lib/python3.6/site-packages/elasticsearch/client/utils.py:76: in _wrapped
    return func(*args, params=params, **kwargs)
.tox/c1/lib/python3.6/site-packages/elasticsearch/client/indices.py:91: in create
    params=params, body=body)
.tox/c1/lib/python3.6/site-packages/elasticsearch/transport.py:314: in perform_request
    status, headers_response, data = connection.perform_request(method, url, params, body, headers=headers, ignore=ignore, timeout=timeout)
.tox/c1/lib/python3.6/site-packages/elasticsearch/connection/http_urllib3.py:163: in perform_request
    self._raise_error(response.status, raw_data)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Urllib3HttpConnection: http://elasticsearch:9200>, status_code = 403
raw_data = '{"error":{"root_cause":[{"type":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow...":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"},"status":403}'

    def _raise_error(self, status_code, raw_data):
        """ Locate appropriate exception and raise it. """
        error_message = raw_data
        additional_info = None
        try:
            if raw_data:
                additional_info = json.loads(raw_data)
                error_message = additional_info.get('error', error_message)
                if isinstance(error_message, dict) and 'type' in error_message:
                    error_message = error_message['type']
        except (ValueError, TypeError) as err:
            logger.warning('Undecodable raw error response from server: %s', err)
    
>       raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
E       elasticsearch.exceptions.AuthorizationException: TransportError(403, 'cluster_block_exception', 'blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];')

.tox/c1/lib/python3.6/site-packages/elasticsearch/connection/base.py:125: AuthorizationException
------------------------------ Captured log setup ------------------------------
WARNING  elasticsearch:base.py:97 PUT http://elasticsearch:9200/test-weko-item-v1.0.0 [status:403 request:0.007s]
=================================== FAILURES ===================================
_______________________________ test_get_records _______________________________

es_app = <Flask 'testapp'>
db = <SQLAlchemy engine=sqlite:////code/modules/invenio-oaiserver/test.db>
mock_execute = <function mock_execute.<locals>.factory at 0x7f395da67d08>

    def test_get_records(es_app,db, mock_execute):
        indexes = list()
        for i in range(10):
            indexes.append(
                Index(
                    parent=0,
                    position=i,
                    index_name_english="test_index{}".format(i),
                    index_link_name_english="test_index_link{}".format(i),
                    harvest_public_state=True,
                    public_state=True,
                    public_date=datetime(2100,1,1),
                    browsing_role="3,-99"
                )
            )
        rec_uuid1 = uuid.uuid4()
        identifier1 = PersistentIdentifier.create('doi', "https://doi.org/00001",object_type='rec', object_uuid=rec_uuid1,status=PIDStatus.REGISTERED)
    
        rec_data1={"title":["test_item1"],"path":[1],"_oai":{"id":"oai:test:00001","sets":[]},"relation_version_is_last":"true","control_number":"1"}
        rec1 = RecordMetadata(id=rec_uuid1,json=rec_data1)
        rec_uuid2 = uuid.uuid4()
        rec_data2={"title":["test_item2"],"path":[1000],"_oai":{"id":"oai:test:00002","sets":["12345"]},"relation_version_is_last":"true","control_number":"2"}
        identifier2 = PersistentIdentifier.create('doi', "https://doi.org/00002",object_type='rec', object_uuid=rec_uuid2,status=PIDStatus.REGISTERED)
        rec2 = RecordMetadata(id=rec_uuid2,json=rec_data2)
        db.session.add_all(indexes)
        db.session.add(rec1)
        db.session.add(rec2)
    
        db.session.commit()
    
        es_info = dict(id=str(rec_uuid1),
                           index=current_app.config['INDEXER_DEFAULT_INDEX'],
                           doc_type=current_app.config['INDEXER_DEFAULT_DOCTYPE'])
        body = dict(version=1,
                    version_type="external_gte",
                    body=rec_data1)
>       current_search_client.index(**{**es_info,**body})

tests/test_query.py:112: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.tox/c1/lib/python3.6/site-packages/elasticsearch/client/utils.py:76: in _wrapped
    return func(*args, params=params, **kwargs)
.tox/c1/lib/python3.6/site-packages/elasticsearch/client/__init__.py:300: in index
    _make_path(index, doc_type, id), params=params, body=body)
.tox/c1/lib/python3.6/site-packages/elasticsearch/transport.py:314: in perform_request
    status, headers_response, data = connection.perform_request(method, url, params, body, headers=headers, ignore=ignore, timeout=timeout)
.tox/c1/lib/python3.6/site-packages/elasticsearch/connection/http_urllib3.py:163: in perform_request
    self._raise_error(response.status, raw_data)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Urllib3HttpConnection: http://elasticsearch:9200>, status_code = 403
raw_data = '{"error":{"root_cause":[{"type":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow...":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"},"status":403}'

    def _raise_error(self, status_code, raw_data):
        """ Locate appropriate exception and raise it. """
        error_message = raw_data
        additional_info = None
        try:
            if raw_data:
                additional_info = json.loads(raw_data)
                error_message = additional_info.get('error', error_message)
                if isinstance(error_message, dict) and 'type' in error_message:
                    error_message = error_message['type']
        except (ValueError, TypeError) as err:
            logger.warning('Undecodable raw error response from server: %s', err)
    
>       raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
E       elasticsearch.exceptions.AuthorizationException: TransportError(403, 'cluster_block_exception', 'blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];')

.tox/c1/lib/python3.6/site-packages/elasticsearch/connection/base.py:125: AuthorizationException
------------------------------ Captured log call -------------------------------
WARNING  elasticsearch:base.py:97 PUT http://elasticsearch:9200/test-weko-item-v1.0.0/item-v1.0.0/2fb47f17-1422-4fe4-b9d8-1395cc6e1070?version=1&version_type=external_gte [status:403 request:0.004s]
________________________________ test_getrecord ________________________________

app = <Flask 'testapp'>
records = [(<PersistentIdentifier recid:1 / rec:02187805-36b2-4611-9c8a-aca46c8b4f8a (R)>, <invenio_oaiserver.provider.OAIIDProv...etadata': {}}}, '_oai': {'sets': []}, 'item_type_id': '1', 'control_number': '6'}, {'title': 'publis_status_-1'}), ...]
item_type = {}
mock_execute = <function mock_execute.<locals>.factory at 0x7f395d1c5f28>
db = <SQLAlchemy engine=sqlite:////code/modules/invenio-oaiserver/test.db>
mocker = <pytest_mock.plugin.MockerFixture object at 0x7f395d157f60>

    def test_getrecord(app, records, item_type, mock_execute, db, mocker):
        """Test of method which creates OAI-PMH response for verb GetRecord."""
        with app.app_context():
            identify = Identify(
                outPutSetting=True
            )
            index_metadata = {
                "id": 1557819692844,
                "parent": 0,
                "position": 0,
                "index_name": "コンテンツタイプ (Contents Type)",
                "index_name_english": "Contents Type",
                "index_link_name": "",
                "index_link_name_english": "New Index",
                "index_link_enabled": False,
                "more_check": False,
                "display_no": 5,
                "harvest_public_state": True,
                "display_format": 1,
                "image_name": "",
                "public_state": True,
                "recursive_public_state": True,
                "rss_status": False,
                "coverpage_state": False,
                "recursive_coverpage_check": False,
                "browsing_role": "3,-98,-99",
                "recursive_browsing_role": False,
                "contribute_role": "1,2,3,4,-98,-99",
                "recursive_contribute_role": False,
                "browsing_group": "",
                "recursive_browsing_group": False,
                "recursive_contribute_group": False,
                "owner_user_id": 1,
                "item_custom_sort": {"2": 1}
            }
            index = Index(**index_metadata)
            mapping = Mapping.create(
                item_type_id=item_type.id,
                mapping={}
            )
            with db.session.begin_nested():
                db.session.add(identify)
                db.session.add(index)
            dummy_data = {
                "hits": {
                    "total": 3,
                    "hits": [
                        {
                            "_source": {
                                "_oai": {"id": str(records[0][0])},
                                "_updated": "2022-01-01T10:10:10"
                            },
                            "_id": records[0][2].id,
                        },
                        {
                            "_source": {
                                "_oai": {"id": str(records[1][0])},
                                "_updated": "2022-01-01T10:10:10"
                            },
                            "_id": records[1][2].id,
                        },
                        {
                            "_source": {
                                "_oai": {"id": str(records[2][0])},
                                "_updated": "2022-01-01T10:10:10"
                            },
                            "_id": records[2][2].id,
                        },
                    ]
                }
            }
            kwargs = dict(
                metadataPrefix="jpcoar_1.0",
                verb="GetRecord",
                identifier=str(records[0][0])
            )
            ns = {"root_name": "jpcoar", "namespaces":{'': 'https://github.com/JPCOAR/schema/blob/master/1.0/',
                  'dc': 'http://purl.org/dc/elements/1.1/', 'xs': 'http://www.w3.org/2001/XMLSchema',
                  'rdf': 'http://www.w3.org/1999/02/22-rdf-syntax-ns#', 'xml': 'http://www.w3.org/XML/1998/namespace',
                  'dcndl': 'http://ndl.go.jp/dcndl/terms/', 'oaire': 'http://namespace.openaire.eu/schema/oaire/',
                  'jpcoar': 'https://github.com/JPCOAR/schema/blob/master/1.0/',
                  'dcterms': 'http://purl.org/dc/terms/', 'datacite': 'https://schema.datacite.org/meta/kernel-4/',
                  'rioxxterms': 'http://www.rioxx.net/schema/v2.0/rioxxterms/'}}
            mocker.patch("invenio_oaiserver.response.to_utc",side_effect=lambda x:x)
            mocker.patch("weko_index_tree.utils.get_user_groups",return_value=[])
            mocker.patch("weko_index_tree.utils.check_roles",return_value=True)
            mocker.patch("invenio_oaiserver.response.get_identifier",return_value=None)
            mocker.patch("weko_schema_ui.schema.cache_schema",return_value=ns)
            with patch("invenio_oaiserver.query.OAIServerSearch.execute",return_value=mock_execute(dummy_data)):
                # not identify
                with patch("invenio_oaiserver.response.OaiIdentify.get_all",return_value=None):
                    res = getrecord(**kwargs)
                    assert res.xpath("/x:OAI-PMH/x:error",namespaces=NAMESPACES)[0].attrib["code"] == "idDoesNotExist"
    
                # output setting of identity = false
                identify = Identify(
                    outPutSetting=False
                )
                with patch("invenio_oaiserver.response.OaiIdentify.get_all",return_value=identify):
                    res = getrecord(**kwargs)
                    assert res.xpath("/x:OAI-PMH/x:error",namespaces=NAMESPACES)[0].attrib["code"] == "idDoesNotExist"
    
                # harvest setting of index = private
                with patch("invenio_oaiserver.response.is_output_harvest",return_value=HARVEST_PRIVATE):
                    res = getrecord(**kwargs)
                    assert res.xpath("/x:OAI-PMH/x:error",namespaces=NAMESPACES)[0].attrib["code"] == "idDoesNotExist"
    
            # path is none
            dummy_data = {
                "hits": {
                    "total": 1,
                    "hits": [
                        {
                            "_source": {
                                "_oai": {"id": str(records[2][0])},
                                "_updated": "2022-01-01T10:10:10"
                            },
                            "_id": records[1][2].id,
                        }
                    ]
                }
            }
            kwargs = dict(
                metadataPrefix='jpcoar_1.0',
                verb="GetRecord",
                identifier=str(records[1][0])
            )
            with patch("invenio_oaiserver.query.OAIServerSearch.execute",return_value=mock_execute(dummy_data)):
                res = getrecord(**kwargs)
                identifier = res.xpath(
                    '/x:OAI-PMH/x:GetRecord/x:record/x:header/x:identifier/text()',
                    namespaces=NAMESPACES)
                assert identifier == [str(records[1][0])]
                header = res.xpath(
                    '/x:OAI-PMH/x:GetRecord/x:record/x:header[@status="deleted"]',
                    namespaces=NAMESPACES)
                assert len(header) == 1
    
            # publish_status = 0 (public item)
            dummy_data = {
                "hits": {
                    "total": 1,
                    "hits": [
                        {
                            "_source": {
                                "_oai": {"id": str(records[2][0])},
                                "_updated": "2022-01-01T10:10:10"
                            },
                            "_id": records[2][2].id,
                        }
                    ]
                }
            }
            kwargs = dict(
                metadataPrefix='jpcoar_1.0',
                verb="GetRecord",
                identifier=str(records[2][0])
            )
            with patch("invenio_oaiserver.query.OAIServerSearch.execute",return_value=mock_execute(dummy_data)):
                # not etree_record.get("system_identifier_doi")
                with patch("invenio_oaiserver.response.is_exists_doi",return_value=False):
                    res = getrecord(**kwargs)
                    identifier = res.xpath(
                        '/x:OAI-PMH/x:GetRecord/x:record/x:header/x:identifier/text()',
                        namespaces=NAMESPACES)
                    assert identifier == [str(records[2][0])]
                    datestamp = res.xpath(
                        '/x:OAI-PMH/x:GetRecord/x:record/x:header/x:datestamp/text()',
                        namespaces=NAMESPACES)
                    assert datestamp == [datetime_to_datestamp(records[2][0].updated)]
    
                    # private index
                    with patch("invenio_oaiserver.response.is_output_harvest",return_value=PRIVATE_INDEX):
                        res = getrecord(**kwargs)
                        identifier = res.xpath(
                            '/x:OAI-PMH/x:GetRecord/x:record/x:header/x:identifier/text()',
                            namespaces=NAMESPACES)
                        assert identifier == [str(records[2][0])]
                        header = res.xpath(
                            '/x:OAI-PMH/x:GetRecord/x:record/x:header[@status="deleted"]',
                            namespaces=NAMESPACES)
                        assert len(header) == 1
    
                kwargs = dict(
                    metadataPrefix='jpcoar_1.0',
                    verb="GetRecord",
                    identifier=str(records[3][0])
                )
                # etree_record.get("system_identifier_doi")
                with patch("invenio_oaiserver.response.is_exists_doi",return_value=True):
                    res = getrecord(**kwargs)
                    identifier = res.xpath(
                        '/x:OAI-PMH/x:GetRecord/x:record/x:header/x:identifier/text()',
                        namespaces=NAMESPACES)
>                   assert identifier == [str(records[3][0])]
E                   AssertionError: assert [] == ['<PersistentIdentifier recid:4 / rec:54882b48-e825-4604-9d7c-4f64d55124cd '\n '(R)>']
E                     Right contains one more item: '<PersistentIdentifier recid:4 / rec:54882b48-e825-4604-9d7c-4f64d55124cd (R)>'
E                     Full diff:
E                       [
E                     +  ,
E                     -  '<PersistentIdentifier recid:4 / rec:54882b48-e825-4604-9d7c-4f64d55124cd '
E                     -  '(R)>',
E                       ]

tests/test_response.py:263: AssertionError
------------------------------ Captured log call -------------------------------
ERROR    testapp:utils.py:82 'NoneType' object has no attribute 'app'
ERROR    testapp:utils.py:82 'NoneType' object has no attribute 'app'
ERROR    testapp:utils.py:82 'NoneType' object has no attribute 'app'
ERROR    testapp:utils.py:82 'NoneType' object has no attribute 'app'
ERROR    testapp:utils.py:82 'NoneType' object has no attribute 'app'
ERROR    testapp:utils.py:82 'NoneType' object has no attribute 'app'
ERROR    testapp:utils.py:82 'NoneType' object has no attribute 'app'
___________________ test_check_correct_system_props_mapping ____________________

app = <Flask 'testapp'>
db = <SQLAlchemy engine=sqlite:////code/modules/invenio-oaiserver/test.db>

    def test_check_correct_system_props_mapping(app,db):
        obj_uuid = uuid.uuid4()
        item_metadata1 = ItemMetadata(id=obj_uuid,item_type_id=1,json={})
        mapping_data = {
            "ITEM1":{"jpcoar_mapping":{"item1":{"subitem1_1":"item1.subitem1_1","subitem2_1":"item1.subitem2_1"}}},
            "ITEM2":{"jpcoar_mapping":{"item2":{"subitem1_2":{"subitem1_1_2":"item2.subitem1_2.subitem1_1_2"},"subitem2_2":{"subitem1_2_2":"item2.subitem2_2.subitem1_2_2"}}}}
        }
        mapping1 = ItemTypeMapping(item_type_id=1,mapping=mapping_data)
        db.session.add_all([item_metadata1,mapping1])
        db.session.commit()
        # item_map
        #{'item1.subitem1_1': 'ITEM1.item1.subitem1_1',
        # 'item1.subitem2_1': 'ITEM1.item1.subitem2_1',
        # 'item2.subitem1_2.subitem1_1_2': 'ITEM2.item2.subitem1_2.subitem1_1_2',
        # 'item2.subitem2_2.subitem1_2_2': 'ITEM2.item2.subitem2_2.subitem1_2_2'}
    
        # pass check
        system_mapping_config={"item1.subitem1_1":"ITEM1.item1.subitem1_1","item2.subitem1_2.subitem1_1_2": "ITEM2.item2.subitem1_2.subitem1_1_2"}
>       result = check_correct_system_props_mapping(obj_uuid,system_mapping_config)

tests/test_response.py:1566: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
invenio_oaiserver/response.py:721: in check_correct_system_props_mapping
    item_map = get_mapping(item_type.item_type_id, "jpcoar_mapping")
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

item_type_id = 1, mapping_type = 'jpcoar_mapping'

    def get_mapping(item_type_id, mapping_type):
        """Format itemtype mapping data.
    
        [Key:Schema, Value:ItemId]
        :param item_type_id:
        :param mapping_type:
        :return:
        """
        def get_schema_key_info(schema, parent_key, schema_json={}):
    
            for k, v in schema.items():
                key = parent_key + '.' + k if parent_key else k
                if isinstance(v, dict):
                    child_key = pickle.loads(pickle.dumps(key, -1))
                    get_schema_key_info(v, child_key, schema_json)
                else:
                    schema_json[key] = v
    
            return schema_json
    
        schema = {}
        item_type_mapping = Mapping.get_record(item_type_id)
>       item_type_list = ItemTypes.get_by_id(item_type_id).render.get('table_row')
E       AttributeError: 'NoneType' object has no attribute 'render'

../weko-records/weko_records/serializers/utils.py:62: AttributeError
________________________ test_combine_record_file_urls _________________________

app = <Flask 'testapp'>
db = <SQLAlchemy engine=sqlite:////code/modules/invenio-oaiserver/test.db>
mocker = <pytest_mock.plugin.MockerFixture object at 0x7f395cbb1780>

    def test_combine_record_file_urls(app,db,mocker):
        metadata_formats = {
            'jpcoar_1.0': {'serializer': ('weko_schema_ui.utils:dumps_oai_etree', {'schema_type': 'jpcoar_v1'}), 'namespace': 'https://irdb.nii.ac.jp/schema/jpcoar/1.0/', 'schema': 'https://irdb.nii.ac.jp/schema/jpcoar/1.0/jpcoar_scm.xsd'},
            'jpcoar': {'serializer': ('weko_schema_ui.utils:dumps_oai_etree', {'schema_type': 'jpcoar'}), 'namespace': 'https://irdb.nii.ac.jp/schema/jpcoar/1.0/', 'schema': 'https://irdb.nii.ac.jp/schema/jpcoar/1.0/jpcoar_scm.xsd'}
        }
        mocker.patch("weko_schema_ui.schema.get_oai_metadata_formats",return_value=metadata_formats)
    
        mapping_data1 = {
            "item_1617605131499": {
                "jpcoar_mapping": {
                    "file": {
                        "URI": {
                            "@value": "url.url",
                            "@attributes": {
                                "label": "url.label",
                                "objectType": "url.objectType"
                            }
                        },
                        "date": {
                            "@value": "fileDate.fileDateValue",
                            "@attributes": { "dateType": "fileDate.fileDateType" }
                        },
                        "extent": { "@value": "filesize.value" },
                        "version": { "@value": "version" },
                        "mimeType": { "@value": "format" }
                    }
                },
                "jpcoar_v1_mapping": {
                    "file": {
                        "URI": {
                            "@value": "url.url",
                            "@attributes": {
                                "label": "url.label",
                                "objectType": "url.objectType"
                            }
                        },
                        "date": {
                            "@value": "fileDate.fileDateValue",
                            "@attributes": { "dateType": "fileDate.fileDateType" }
                        },
                        "extent": { "@value": "filesize.value" },
                        "version": { "@value": "version" },
                        "mimeType": { "@value": "format" }
                    }
                }
            }
        }
        mapping_data2 = { # len(file_keys) != 3
            "item_1617605131499": {
                "jpcoar_mapping": {
                    "file": {
                        "URI": {
                            "@value": "url",
                            "@attributes": {
                                "label": "url.label",
                                "objectType": "url.objectType"
                            }
                        },
                        "date": {
                            "@value": "fileDate.fileDateValue",
                            "@attributes": { "dateType": "fileDate.fileDateType" }
                        },
                        "extent": { "@value": "filesize.value" },
                        "version": { "@value": "version" },
                        "mimeType": { "@value": "format" }
                    }
                },
                "jpcoar_v1_mapping": {
                    "file": {
                        "URI": {
                            "@value": "url.url",
                            "@attributes": {
                                "label": "url.label",
                                "objectType": "url.objectType"
                            }
                        },
                        "date": {
                            "@value": "fileDate.fileDateValue",
                            "@attributes": { "dateType": "fileDate.fileDateType" }
                        },
                        "extent": { "@value": "filesize.value" },
                        "version": { "@value": "version" },
                        "mimeType": { "@value": "format" }
                    }
                }
            }
        }
        mapping1 = ItemTypeMapping(item_type_id=1,mapping=mapping_data1)
        mapping2 = ItemTypeMapping(item_type_id=2,mapping=mapping_data2)
        mapping3 = ItemTypeMapping(item_type_id=3,mapping={})
    
        db.session.add_all([mapping1,mapping2,mapping3])
    
        record_data1 = {
            "recid":"1",
            "item_1617605131499":{
                "attribute_name":"File",
                "attribute_type":"file",
                "attribute_value_mlt":[ # attribute_value_mlt is list
                    { # not exist filename
                        "url":{"url":"https://weko3.example.org/record/1/files/sample_file1"}
                    },
                    { # not exist file_keys[1]
                        "filename":"sample_file2"
                    },
                    { # exist file_keys[1]
                        "url":{"url":"https://weko3.example.org/record/1/files/sample_file3"},
                        "filename":"sample_file3"
                    }
                ]
            }
        }
        record_data2 = {
            "recid":"2",
            "item_1617605131499":{
                "attribute_name":"File",
                "attribute_type":"file",
                "attribute_value_mlt":{ # attribute_value_mlt is dict
                    "filename":"sample_file" # not exist file_keys[1]
                }
            }
        }
        record_data3 = {
            "recid":"3",
            "item_1617605131499":{
                "attribute_name":"File",
                "attribute_type":"file",
                "attribute_value_mlt":{ # attribute_value_mlt is dict
                    # exist file_keys[1]
                    "url":{"url":"https://weko3.example.org/record/3/files/sample_file"},
                    "filename":"sample_file"
                }
            }
        }
        record_data4 = {
            "recid":"4",
            "item_1617605131499":{
                "attribute_name":"File",
                "attribute_type":"file",
                "attribute_value_mlt":{ # attribute_value_mlt is dict
                    # not exist filename
                    "url":{"url":"https://weko3.example.org/record/4/files/sample_file"},
                }
            }
        }
        record_data5 = {
            "recid":"5",
            "item_1617605131499":{
                "attribute_name":"File",
                "attribute_type":"file",
                "attribute_value_mlt":{ # attribute_value_mlt is dict
                    # not exist filename
                    "url":"https://weko3.example.org/record/4/files/sample_file",
                }
            }
        }
    
        rec_uuid1 = uuid.uuid4()
        record1 = WekoRecord.create(record_data1, id_=rec_uuid1)
        item_metadata1 = ItemMetadata(id=rec_uuid1,item_type_id=1,json={})
    
        rec_uuid2 = uuid.uuid4()
        record2 = WekoRecord.create(record_data2, id_=rec_uuid2)
        item_metadata2 = ItemMetadata(id=rec_uuid2,item_type_id=1,json={})
    
        rec_uuid3 = uuid.uuid4()
        record3 = WekoRecord.create(record_data3, id_=rec_uuid3)
        item_metadata3 = ItemMetadata(id=rec_uuid3,item_type_id=1,json={})
    
        rec_uuid4 = uuid.uuid4()
        record4 = WekoRecord.create(record_data4, id_=rec_uuid4)
        item_metadata4 = ItemMetadata(id=rec_uuid4,item_type_id=1,json={})
    
        rec_uuid5 = uuid.uuid4()
        record5 = WekoRecord.create(record_data5, id_=rec_uuid5)
        item_metadata5 = ItemMetadata(id=rec_uuid5,item_type_id=2,json={}
                                      )
        rec_uuid6 = uuid.uuid4()
        item_metadata6 = ItemMetadata(id=rec_uuid6,item_type_id=3,json={})
    
        db.session.add_all([item_metadata1,item_metadata2,item_metadata3,item_metadata4,item_metadata5,item_metadata6])
        db.session.commit()
    
        with app.test_request_context():
            # not item_map
>           result = combine_record_file_urls(record1,rec_uuid6,"jpcoar_1.0")

tests/test_response.py:1765: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
invenio_oaiserver/response.py:746: in combine_record_file_urls
    "{}_mapping".format(mapping_type))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

item_type_id = 3, mapping_type = 'jpcoar_v1_mapping'

    def get_mapping(item_type_id, mapping_type):
        """Format itemtype mapping data.
    
        [Key:Schema, Value:ItemId]
        :param item_type_id:
        :param mapping_type:
        :return:
        """
        def get_schema_key_info(schema, parent_key, schema_json={}):
    
            for k, v in schema.items():
                key = parent_key + '.' + k if parent_key else k
                if isinstance(v, dict):
                    child_key = pickle.loads(pickle.dumps(key, -1))
                    get_schema_key_info(v, child_key, schema_json)
                else:
                    schema_json[key] = v
    
            return schema_json
    
        schema = {}
        item_type_mapping = Mapping.get_record(item_type_id)
>       item_type_list = ItemTypes.get_by_id(item_type_id).render.get('table_row')
E       AttributeError: 'NoneType' object has no attribute 'render'

../weko-records/weko_records/serializers/utils.py:62: AttributeError
=============================== warnings summary ===============================
.tox/c1/lib/python3.6/site-packages/click/core.py:1259
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/click/core.py:1259: DeprecationWarning: invalid escape sequence \s
    """

.tox/c1/lib/python3.6/site-packages/click/_compat.py:14
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/click/_compat.py:14: DeprecationWarning: invalid escape sequence \[
    _ansi_re = re.compile('\033\[((?:\d|;)*)([a-zA-Z])')

.tox/c1/lib/python3.6/site-packages/click/decorators.py:111
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/click/decorators.py:111: DeprecationWarning: invalid escape sequence \s
    """

.tox/c1/lib/python3.6/site-packages/blinker/base.py:110
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/blinker/base.py:110: DeprecationWarning: invalid escape sequence \*
    """

.tox/c1/lib/python3.6/site-packages/blinker/base.py:180
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/blinker/base.py:180: DeprecationWarning: invalid escape sequence \*
    """

.tox/c1/lib/python3.6/site-packages/blinker/base.py:252
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/blinker/base.py:252: DeprecationWarning: invalid escape sequence \*
    """

.tox/c1/lib/python3.6/site-packages/babel/core.py:1130
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/babel/core.py:1130: DeprecationWarning: invalid escape sequence \s
    """

.tox/c1/lib/python3.6/site-packages/babel/dates.py:1158
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/babel/dates.py:1158: DeprecationWarning: invalid escape sequence \d
    numbers = re.findall('(\d+)', string)

.tox/c1/lib/python3.6/site-packages/babel/dates.py:1201
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/babel/dates.py:1201: DeprecationWarning: invalid escape sequence \d
    numbers = re.findall('(\d+)', string)

.tox/c1/lib/python3.6/site-packages/babel/localtime/_unix.py:103
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/babel/localtime/_unix.py:103: DeprecationWarning: invalid escape sequence \s
    zone_re = re.compile('\s*ZONE\s*=\s*\"')

.tox/c1/lib/python3.6/site-packages/babel/localtime/_unix.py:104
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/babel/localtime/_unix.py:104: DeprecationWarning: invalid escape sequence \s
    timezone_re = re.compile('\s*TIMEZONE\s*=\s*\"')

.tox/c1/lib/python3.6/site-packages/arrow/parser.py:24
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/arrow/parser.py:24: DeprecationWarning: invalid escape sequence \[
    _ESCAPE_RE = re.compile('\[[^\[\]]*\]')

.tox/c1/lib/python3.6/site-packages/arrow/parser.py:26
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/arrow/parser.py:26: DeprecationWarning: invalid escape sequence \d
    _ONE_OR_MORE_DIGIT_RE = re.compile('\d+')

.tox/c1/lib/python3.6/site-packages/arrow/parser.py:27
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/arrow/parser.py:27: DeprecationWarning: invalid escape sequence \d
    _ONE_OR_TWO_DIGIT_RE = re.compile('\d{1,2}')

.tox/c1/lib/python3.6/site-packages/arrow/parser.py:28
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/arrow/parser.py:28: DeprecationWarning: invalid escape sequence \d
    _FOUR_DIGIT_RE = re.compile('\d{4}')

.tox/c1/lib/python3.6/site-packages/arrow/parser.py:29
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/arrow/parser.py:29: DeprecationWarning: invalid escape sequence \d
    _TWO_DIGIT_RE = re.compile('\d{2}')

.tox/c1/lib/python3.6/site-packages/arrow/parser.py:30
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/arrow/parser.py:30: DeprecationWarning: invalid escape sequence \-
    _TZ_RE = re.compile('[+\-]?\d{2}:?(\d{2})?')

.tox/c1/lib/python3.6/site-packages/arrow/parser.py:31
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/arrow/parser.py:31: DeprecationWarning: invalid escape sequence \w
    _TZ_NAME_RE = re.compile('\w[\w+\-/]+')

.tox/c1/lib/python3.6/site-packages/arrow/parser.py:49
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/arrow/parser.py:49: DeprecationWarning: invalid escape sequence \d
    'X': re.compile('\d+'),

.tox/c1/lib/python3.6/site-packages/arrow/parser.py:310
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/arrow/parser.py:310: DeprecationWarning: invalid escape sequence \-
    _TZINFO_RE = re.compile('([+\-])?(\d\d):?(\d\d)?')

.tox/c1/lib/python3.6/site-packages/passlib/context.py:231
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/passlib/context.py:231: DeprecationWarning: invalid escape sequence \;
    elif any(c in source for c in "\n\r\t") or not source.strip(" \t./\;:"):

.tox/c1/lib/python3.6/site-packages/passlib/context.py:2240
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/passlib/context.py:2240: DeprecationWarning: invalid escape sequence \*
    """

.tox/c1/lib/python3.6/site-packages/passlib/context.py:2323
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/passlib/context.py:2323: DeprecationWarning: invalid escape sequence \*
    """

.tox/c1/lib/python3.6/site-packages/passlib/context.py:2406
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/passlib/context.py:2406: DeprecationWarning: invalid escape sequence \*
    """

.tox/c1/lib/python3.6/site-packages/elasticsearch_dsl/document.py:215
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/elasticsearch_dsl/document.py:215: DeprecationWarning: invalid escape sequence \s
    """

.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:223
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:223: DeprecationWarning: invalid escape sequence \s
    ('\s+', Text),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:316
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:316: DeprecationWarning: invalid escape sequence \]
    (r'(/)(\[' + _dot + '*?\])(' + _dot + r'*\n)',

.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:333
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:333: DeprecationWarning: invalid escape sequence \}
    (r'\{(,\n|' + _dot + ')*?\}', using(RubyLexer)),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:334
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:334: DeprecationWarning: invalid escape sequence \]
    (r'\[' + _dot + '*?\]', using(RubyLexer)),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:343
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:343: DeprecationWarning: invalid escape sequence \}
    (r'(#\{)(' + _dot + '*?)(\})',

.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:376
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:376: DeprecationWarning: invalid escape sequence \}
    (r'(#\{)(' + _dot + '*?)(\})',

.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:425
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:425: DeprecationWarning: invalid escape sequence \]
    (r'(/)(\[' + _dot + '*?\])(' + _dot + r'*\n)',

.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:445
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:445: DeprecationWarning: invalid escape sequence \}
    (r'\{(,\n|' + _dot + ')*?\}', using(ScalaLexer)),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:446
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:446: DeprecationWarning: invalid escape sequence \]
    (r'\[' + _dot + '*?\]', using(ScalaLexer)),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:455
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:455: DeprecationWarning: invalid escape sequence \}
    (r'(#\{)(' + _dot + '*?)(\})',

.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:488
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:488: DeprecationWarning: invalid escape sequence \}
    (r'(#\{)(' + _dot + '*?)(\})',

.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:533
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:533: DeprecationWarning: invalid escape sequence \]
    (r'(/)(\[' + _dot + '*?\])(' + _dot + r'*\n)',

.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:554
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:554: DeprecationWarning: invalid escape sequence \}
    (r'\{(,\n|' + _dot + ')*?\}', using(ScalaLexer)),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:555
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:555: DeprecationWarning: invalid escape sequence \]
    (r'\[' + _dot + '*?\]', using(ScalaLexer)),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:564
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:564: DeprecationWarning: invalid escape sequence \}
    (r'(#\{)(' + _dot + '*?)(\})',

.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:597
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/html.py:597: DeprecationWarning: invalid escape sequence \}
    (r'(#\{)(' + _dot + '*?)(\})',

.tox/c1/lib/python3.6/site-packages/pygments/lexers/javascript.py:538
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/javascript.py:538: DeprecationWarning: invalid escape sequence \s
    if re.search('^(import.+(from\s+)?["\']|'

.tox/c1/lib/python3.6/site-packages/pygments/lexers/javascript.py:539
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/javascript.py:539: DeprecationWarning: invalid escape sequence \s
    '(export\s*)?(interface|class|function)\s+)',

.tox/c1/lib/python3.6/site-packages/pygments/lexers/javascript.py:1018
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/javascript.py:1018: DeprecationWarning: invalid escape sequence \s
    if re.search('^\s*@import\s+[<"]', text, re.MULTILINE):

.tox/c1/lib/python3.6/site-packages/pygments/lexers/jvm.py:260
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/jvm.py:260: DeprecationWarning: invalid escape sequence \d
    letter_letter_digit = u'%s(?:%s|\d)*' % (letter, letter)

.tox/c1/lib/python3.6/site-packages/pygments/lexers/jvm.py:692
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/jvm.py:692: DeprecationWarning: invalid escape sequence \w
    u'(?![\w!:?])', Name.Function),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/jvm.py:1261
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/jvm.py:1261: DeprecationWarning: invalid escape sequence \w
    ('`?[a-zA-Z_][\w$]*', Name),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/ruby.py:406
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/ruby.py:406: DeprecationWarning: invalid escape sequence \(
    _prompt_re = re.compile('irb\([a-zA-Z_]\w*\):\d{3}:\d+[>*"\'] '

.tox/c1/lib/python3.6/site-packages/pygments/lexers/ruby.py:407
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/ruby.py:407: DeprecationWarning: invalid escape sequence \?
    '|>> |\?> ')

.tox/c1/lib/python3.6/site-packages/pygments/lexers/ruby.py:501
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/ruby.py:501: DeprecationWarning: invalid escape sequence \w
    ('[A-Z]\w*', Name.Constant),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/ruby.py:502
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/ruby.py:502: DeprecationWarning: invalid escape sequence \w
    ('@[a-zA-Z_]\w*', Name.Variable.Instance),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/ruby.py:503
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/ruby.py:503: DeprecationWarning: invalid escape sequence \w
    ('@@[a-zA-Z_]\w*', Name.Variable.Class),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/ruby.py:505
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/ruby.py:505: DeprecationWarning: invalid escape sequence \w
    ('[a-zA-Z_]\w*', Name),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/actionscript.py:128
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/actionscript.py:128: DeprecationWarning: invalid escape sequence \.
    typeidentifier = identifier + '(?:\.<\w+>)?'

.tox/c1/lib/python3.6/site-packages/pygments/lexers/actionscript.py:235
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/actionscript.py:235: DeprecationWarning: invalid escape sequence \s
    ('\s+', Text),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/php.py:176
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/php.py:176: DeprecationWarning: invalid escape sequence \}
    (r'\$\{\$+' + _ident_inner + '\}', Name.Variable),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/php.py:217
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/php.py:217: DeprecationWarning: invalid escape sequence \[
    (r'\$' + _ident_inner + '(\[\S+?\]|->' + _ident_inner + ')?',

.tox/c1/lib/python3.6/site-packages/pygments/lexers/webmisc.py:441
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/webmisc.py:441: DeprecationWarning: invalid escape sequence \(
    (r'(' + qname + ')(\()?', bygroups(Name, Punctuation), 'operator'),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/webmisc.py:646
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/webmisc.py:646: DeprecationWarning: invalid escape sequence \*
    (ncname + ':\*', Name, 'operator'),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/webmisc.py:647
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/webmisc.py:647: DeprecationWarning: invalid escape sequence \*
    ('\*:'+ncname, Name.Tag, 'operator'),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/webmisc.py:648
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/webmisc.py:648: DeprecationWarning: invalid escape sequence \*
    ('\*', Name.Tag, 'operator'),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/webmisc.py:875
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/webmisc.py:875: DeprecationWarning: invalid escape sequence \`
    """

.tox/c1/lib/python3.6/site-packages/pygments/lexers/lisp.py:142
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/lisp.py:142: DeprecationWarning: invalid escape sequence \(
    ("(?<=\()(%s)" % '|'.join(re.escape(entry) + ' ' for entry in builtins),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/lisp.py:324
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/lisp.py:324: DeprecationWarning: invalid escape sequence \s
    (r'#+nil' + terminated + '\s*\(', Comment.Preproc, 'commented-form'),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/lisp.py:336
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/lisp.py:336: DeprecationWarning: invalid escape sequence \*
    (r'\*' + symbol + '\*', Name.Variable.Global),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/lisp.py:2157
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/lisp.py:2157: DeprecationWarning: invalid escape sequence \*
    (r'\*' + symbol + '\*', Name.Variable.Global),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/python.py:183
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/python.py:183: DeprecationWarning: invalid escape sequence \w
    ('[a-zA-Z_]\w*', Name),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/python.py:187
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/python.py:187: DeprecationWarning: invalid escape sequence \w
    ('[a-zA-Z_]\w*', Name.Function, '#pop'),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/python.py:191
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/python.py:191: DeprecationWarning: invalid escape sequence \w
    ('[a-zA-Z_]\w*', Name.Class, '#pop')

.tox/c1/lib/python3.6/site-packages/pygments/lexers/python.py:268
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/python.py:268: DeprecationWarning: invalid escape sequence \w
    '((\w+)((\.\w+)|(\[[^\]]+\]))*)?'  # field name

.tox/c1/lib/python3.6/site-packages/pygments/lexers/python.py:269
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/python.py:269: DeprecationWarning: invalid escape sequence \!
    '(\![sra])?'                       # conversion

.tox/c1/lib/python3.6/site-packages/pygments/lexers/python.py:270
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/python.py:270: DeprecationWarning: invalid escape sequence \:
    '(\:(.?[<>=\^])?[-+ ]?#?0?(\d+)?,?(\.\d+)?[E-GXb-gnosx%]?)?'

.tox/c1/lib/python3.6/site-packages/pygments/lexers/python.py:271
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/python.py:271: DeprecationWarning: invalid escape sequence \}
    '\}', String.Interpol),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/python.py:674
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/python.py:674: DeprecationWarning: invalid escape sequence \w
    ('[a-zA-Z_]\w*', Name),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/python.py:677
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/python.py:677: DeprecationWarning: invalid escape sequence \w
    ('[a-zA-Z_]\w*', Name.Function, '#pop')

.tox/c1/lib/python3.6/site-packages/pygments/lexers/python.py:694
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/python.py:694: DeprecationWarning: invalid escape sequence \w
    ('[a-zA-Z_]\w*', Name.Class, '#pop')

.tox/c1/lib/python3.6/site-packages/pygments/lexers/perl.py:211
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/perl.py:211: DeprecationWarning: invalid escape sequence \s
    if re.search('(?:my|our)\s+[$@%(]', text):

.tox/c1/lib/python3.6/site-packages/pygments/lexers/perl.py:229
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/perl.py:229: DeprecationWarning: invalid escape sequence \w
    PERL6_IDENTIFIER_RANGE = "['\w:-]"

.tox/c1/lib/python3.6/site-packages/pygments/lexers/perl.py:498
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/perl.py:498: DeprecationWarning: invalid escape sequence \s
    (r'(regex|token|rule)(?!' + PERL6_IDENTIFIER_RANGE + ')(\s*' + PERL6_IDENTIFIER_RANGE + '+)?',

.tox/c1/lib/python3.6/site-packages/pygments/lexers/perl.py:594
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/perl.py:594: DeprecationWarning: invalid escape sequence \s
    if re.search("(?:my|our|has)\s+(?:" + Perl6Lexer.PERL6_IDENTIFIER_RANGE +

.tox/c1/lib/python3.6/site-packages/pygments/lexers/perl.py:595
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/perl.py:595: DeprecationWarning: invalid escape sequence \s
    "+\s+)?[$@%&(]", text):

.tox/c1/lib/python3.6/site-packages/pygments/lexers/perl.py:601
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/perl.py:601: DeprecationWarning: invalid escape sequence \s
    if re.match('^\s*$', line):

.tox/c1/lib/python3.6/site-packages/pygments/lexers/perl.py:605
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/perl.py:605: DeprecationWarning: invalid escape sequence \s
    if re.match('^\s*(?:use\s+)?v6(?:\.\d(?:\.\d)?)?;', line):

.tox/c1/lib/python3.6/site-packages/pygments/lexers/perl.py:608
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/perl.py:608: DeprecationWarning: invalid escape sequence \s
    class_decl = re.match('^\s*(?:(?P<scope>my|our)\s+)?(?:module|class|role|enum|grammar)', line)

.tox/c1/lib/python3.6/site-packages/pygments/lexers/iolang.py:52
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/iolang.py:52: DeprecationWarning: invalid escape sequence \w
    ('[a-zA-Z_]\w*', Name),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/scripting.py:107
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/scripting.py:107: DeprecationWarning: invalid escape sequence \(
    ('\(', Punctuation, '#pop'),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/scripting.py:699
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/scripting.py:699: DeprecationWarning: invalid escape sequence \(
    ('\(\*', Comment.Multiline, '#push'),

.tox/c1/lib/python3.6/site-packages/pygments/lexers/scripting.py:700
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pygments/lexers/scripting.py:700: DeprecationWarning: invalid escape sequence \*
    ('\*\)', Comment.Multiline, '#pop'),

.tox/c1/lib/python3.6/site-packages/elementpath/tdop_parser.py:434
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/elementpath/tdop_parser.py:434: DeprecationWarning: invalid escape sequence \s
    s.replace(r'\ ', '\s+')

.tox/c1/lib/python3.6/site-packages/elementpath/tdop_parser.py:441
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/elementpath/tdop_parser.py:441: DeprecationWarning: invalid escape sequence \s
    s = '%s\s*%s' % (s[:-4], s[-4:])

.tox/c1/lib/python3.6/site-packages/elementpath/tdop_parser.py:439
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/elementpath/tdop_parser.py:439: DeprecationWarning: invalid escape sequence \s
    s = '%s\s*%s' % (s[:-2], s[-2:])

.tox/c1/lib/python3.6/site-packages/elementpath/xpath1_parser.py:32
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/elementpath/xpath1_parser.py:32: DeprecationWarning: invalid escape sequence \-
    XML_NCNAME_PATTERN = u"[{0}][\-.0-9\u00B7\u0300-\u036F\u203F-\u2040{0}]*".format(XML_NAME_CHARACTER)

.tox/c1/lib/python3.6/site-packages/elementpath/xpath1_parser.py:115
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/elementpath/xpath1_parser.py:115: DeprecationWarning: invalid escape sequence \s
    axis_pattern_template = '\\b%s(?=\s*\\:\\:|\s*\\(\\:.*\\:\\)\s*\\:\\:)'

.tox/c1/lib/python3.6/site-packages/elementpath/xpath1_parser.py:169
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/elementpath/xpath1_parser.py:169: DeprecationWarning: invalid escape sequence \s
    function_pattern_template = '\\b%s(?=\s*\\(|\s*\\(\\:.*\\:\\)\\()'

.tox/c1/lib/python3.6/site-packages/elementpath/xpath2_parser.py:1045
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/elementpath/xpath2_parser.py:1045: DeprecationWarning: invalid escape sequence \s
    pattern='\\battribute(?=\s*\\:\\:|\s*\\(\\:.*\\:\\)\s*\\:\\:|\s*\\(|\s*\\(\\:.*\\:\\)\\()')

.tox/c1/lib/python3.6/site-packages/xmlschema/regex.py:39
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/xmlschema/regex.py:39: DeprecationWarning: invalid escape sequence \-
    u"\-.0-9:A-Z_a-z\u00B7\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u037D\u037F-\u1FFF\u200C-"

.tox/c1/lib/python3.6/site-packages/xmlschema/codepoints.py:31
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/xmlschema/codepoints.py:31: DeprecationWarning: invalid escape sequence \%
    return u'\%s' % unicode_chr(cp)

.tox/c1/lib/python3.6/site-packages/xmlschema/codepoints.py:35
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/xmlschema/codepoints.py:35: DeprecationWarning: invalid escape sequence \%
    start_char = u'\%s' % unicode_chr(cp[0])

.tox/c1/lib/python3.6/site-packages/xmlschema/codepoints.py:40
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/xmlschema/codepoints.py:40: DeprecationWarning: invalid escape sequence \%
    end_char = u'\%s' % unicode_chr(cp[1])

.tox/c1/lib/python3.6/site-packages/validators/url.py:5
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/validators/url.py:5: DeprecationWarning: invalid escape sequence \.
    ip_middle_octet = u"(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5]))"

.tox/c1/lib/python3.6/site-packages/validators/url.py:6
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/validators/url.py:6: DeprecationWarning: invalid escape sequence \.
    ip_last_octet = u"(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))"

.tox/c1/lib/python3.6/site-packages/validators/url.py:13
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/validators/url.py:13: DeprecationWarning: invalid escape sequence \S
    u"(?:\S+(?::\S*)?@)?"

.tox/c1/lib/python3.6/site-packages/validators/url.py:19
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/validators/url.py:19: DeprecationWarning: invalid escape sequence \.
    u"(?:(?:169\.254|192\.168)" + ip_middle_octet + ip_last_octet + u")|"

.tox/c1/lib/python3.6/site-packages/validators/url.py:20
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/validators/url.py:20: DeprecationWarning: invalid escape sequence \.
    u"(?:172\.(?:1[6-9]|2\d|3[0-1])" + ip_middle_octet + ip_last_octet + u"))"

.tox/c1/lib/python3.6/site-packages/validators/url.py:32
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/validators/url.py:32: DeprecationWarning: invalid escape sequence \d
    u"(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])"

.tox/c1/lib/python3.6/site-packages/validators/url.py:39
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/validators/url.py:39: DeprecationWarning: invalid escape sequence \.
    u"(?:\.(?:[a-z\u00a1-\uffff0-9]-?)*[a-z\u00a1-\uffff0-9]+)*"

.tox/c1/lib/python3.6/site-packages/validators/url.py:41
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/validators/url.py:41: DeprecationWarning: invalid escape sequence \.
    u"(?:\.(?:[a-z\u00a1-\uffff]{2,}))"

.tox/c1/lib/python3.6/site-packages/validators/url.py:44
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/validators/url.py:44: DeprecationWarning: invalid escape sequence \d
    u"(?::\d{2,5})?"

.tox/c1/lib/python3.6/site-packages/validators/url.py:46
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/validators/url.py:46: DeprecationWarning: invalid escape sequence \S
    u"(?:/\S*)?"

.tox/c1/lib/python3.6/site-packages/validators/url.py:48
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/validators/url.py:48: DeprecationWarning: invalid escape sequence \?
    u"(?:\?\S*)?"

.tox/c1/lib/python3.6/site-packages/jsonpath_ng/ext/string.py:18
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/jsonpath_ng/ext/string.py:18: DeprecationWarning: invalid escape sequence \(
    SUB = re.compile("sub\(/(.*)/,\s+(.*)\)")

.tox/c1/lib/python3.6/site-packages/jsonpath_ng/ext/string.py:19
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/jsonpath_ng/ext/string.py:19: DeprecationWarning: invalid escape sequence \(
    SPLIT = re.compile("split\((.),\s+(\d+),\s+(\d+|-1)\)")

.tox/c1/lib/python3.6/site-packages/jsonpath_ng/ext/string.py:20
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/jsonpath_ng/ext/string.py:20: DeprecationWarning: invalid escape sequence \(
    STR = re.compile("str\(\)")

.tox/c1/lib/python3.6/site-packages/citeproc/formatter/rst.py:8
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/citeproc/formatter/rst.py:8: DeprecationWarning: invalid escape sequence \*
    text = text.replace('*', '\*')

.tox/c1/lib/python3.6/site-packages/citeproc/formatter/rst.py:9
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/citeproc/formatter/rst.py:9: DeprecationWarning: invalid escape sequence \`
    text = text.replace('`', '\`')

.tox/c1/lib/python3.6/site-packages/citeproc/model.py:671
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/citeproc/model.py:671: DeprecationWarning: invalid escape sequence \d
    m = re.search('\d+', first)

.tox/c1/lib/python3.6/site-packages/citeproc/source/bibtex/bibtex.py:156
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/citeproc/source/bibtex/bibtex.py:156: DeprecationWarning: invalid escape sequence \d
    RE_DAY = '(?P<day>\d+)'

.tox/c1/lib/python3.6/site-packages/citeproc/source/bibtex/bibtex.py:157
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/citeproc/source/bibtex/bibtex.py:157: DeprecationWarning: invalid escape sequence \w
    RE_MONTH = '(?P<month>\w+)'

.tox/c1/lib/python3.6/site-packages/b2handle/clientcredentials.py:102
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/b2handle/clientcredentials.py:102: DeprecationWarning: invalid escape sequence \*
    '''

.tox/c1/lib/python3.6/site-packages/b2handle/handleexceptions.py:128
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/b2handle/handleexceptions.py:128: DeprecationWarning: invalid escape sequence \s
    pat = re.compile('>[\s]+<')

.tox/c1/lib/python3.6/site-packages/past/translation/__init__.py:35
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/past/translation/__init__.py:35: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

.tox/c1/lib/python3.6/site-packages/past/types/oldstr.py:33
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/past/types/oldstr.py:33: DeprecationWarning: invalid escape sequence \d
    """

.tox/c1/lib/python3.6/site-packages/b2handle/handleclient.py:177
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/b2handle/handleclient.py:177: DeprecationWarning: invalid escape sequence \*
    '''

.tox/c1/lib/python3.6/site-packages/b2handle/handleclient.py:197
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/b2handle/handleclient.py:197: DeprecationWarning: invalid escape sequence \*
    '''

.tox/c1/lib/python3.6/site-packages/b2handle/handleclient.py:230
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/b2handle/handleclient.py:230: DeprecationWarning: invalid escape sequence \*
    '''

.tox/c1/lib/python3.6/site-packages/b2handle/handleclient.py:248
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/b2handle/handleclient.py:248: DeprecationWarning: invalid escape sequence \*
    '''

.tox/c1/lib/python3.6/site-packages/webassets/filter/__init__.py:51
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/webassets/filter/__init__.py:51: DeprecationWarning: invalid escape sequence \,
    """

.tox/c1/lib/python3.6/site-packages/webassets/filter/replace.py:20
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/webassets/filter/replace.py:20: DeprecationWarning: invalid escape sequence \s
    """

.tox/c1/bin/bagit.py:926
  /code/modules/invenio-oaiserver/.tox/c1/bin/bagit.py:926: DeprecationWarning: invalid escape sequence \c
    """

.tox/c1/src/pypdf2/PyPDF2/generic.py:348
  /code/modules/invenio-oaiserver/.tox/c1/src/pypdf2/PyPDF2/generic.py:348: DeprecationWarning: invalid escape sequence \c
    b_("c") : b_("\c"),

tests/test_admin.py::test_admin
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pkg_resources/_vendor/packaging/specifiers.py:276: DeprecationWarning: Creating a LegacyVersion has been deprecated and will be removed in the next major release
    DeprecationWarning,

tests/test_admin.py: 420 warnings
tests/test_api.py: 280 warnings
tests/test_fetcher.py: 140 warnings
tests/test_invenio_oaiserver.py: 420 warnings
tests/test_minters.py: 140 warnings
tests/test_models.py: 280 warnings
tests/test_percolator.py: 1024 warnings
tests/test_provider.py: 140 warnings
tests/test_query.py: 420 warnings
tests/test_receivers.py: 560 warnings
tests/test_response.py: 2660 warnings
tests/test_resumption_token.py: 280 warnings
tests/test_tasks.py: 420 warnings
tests/test_utils.py: 700 warnings
tests/test_verbs.py: 2240 warnings
tests/test_views_server.py: 560 warnings
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/pkg_resources/_vendor/packaging/version.py:114: DeprecationWarning: Creating a LegacyVersion has been deprecated and will be removed in the next major release
    DeprecationWarning,

tests/test_admin.py::test_admin
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/sickle/utils.py:20: DeprecationWarning: invalid escape sequence \{
    return re.search('(\{.*\})', element.tag).group(1)

tests/test_admin.py::test_OAISetModelView
tests/test_admin.py::test_IdentifyModelView
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/flask/sessions.py:211: UserWarning: "app" is not a valid cookie domain, it must contain a ".". Add an entry to your hosts file, for example "app.localdomain", and use that instead.
    ' "{rv}.localdomain", and use that instead.'.format(rv=rv)

tests/test_invenio_oaiserver.py::test_init
tests/test_percolator.py::test_build_cache
  /code/modules/invenio-oaiserver/invenio_oaiserver/ext.py:135: UserWarning: Please specify the OAISERVER_ID_PREFIX configuration.default value is: oai:d722048c6002:recid/
    app.config.get('OAISERVER_ID_PREFIX')))

tests/test_invenio_oaiserver.py::test_AppState
tests/test_percolator.py::test_build_cache
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/flask_caching/__init__.py:241: DeprecationWarning: Using the initialization functions in flask_caching.backend is deprecated.  Use the a full path to backend classes directly.
    category=DeprecationWarning,

tests/test_percolator.py::test_build_cache
  /code/modules/invenio-oaiserver/.tox/c1/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py:794: FSADeprecationWarning: SQLALCHEMY_TRACK_MODIFICATIONS adds significant overhead and will be disabled by default in the future.  Set it to True or False to suppress this warning.
    'SQLALCHEMY_TRACK_MODIFICATIONS adds significant overhead and '

tests/test_percolator.py::test_get_record_sets
  /code/modules/invenio-oaiserver/tests/test_percolator.py:431: DeprecationWarning: generator 'get_record_sets' raised StopIteration
    result = [r for r in result]

tests/test_query.py::test_get_affected_records
  /code/modules/invenio-oaiserver/tests/test_query.py:54: DeprecationWarning: generator 'get_affected_records' raised StopIteration
    for i in result:

tests/test_response.py::test_envelope
  /code/modules/invenio-oaiserver/tests/test_response.py:1080: FutureWarning: The behavior of this method will change in future versions. Use specific 'len(elem)' or 'elem is not None' test instead.
    assert oaipmh

tests/test_response.py::test_envelope
  /code/modules/invenio-oaiserver/tests/test_response.py:1101: FutureWarning: The behavior of this method will change in future versions. Use specific 'len(elem)' or 'elem is not None' test instead.
    assert oaipmh

-- Docs: https://docs.pytest.org/en/stable/warnings.html

---------- coverage: platform linux, python 3.6.15-final-0 -----------
Name                                    Stmts   Miss Branch BrPart  Cover
-------------------------------------------------------------------------
invenio_oaiserver/__init__.py               6      0      0      0   100%
invenio_oaiserver/admin.py                 39      0      6      0   100%
invenio_oaiserver/api.py                   12      0      2      0   100%
invenio_oaiserver/config.py                25      0      0      0   100%
invenio_oaiserver/errors.py                 4      0      4      0   100%
invenio_oaiserver/ext.py                   61      0     24      0   100%
invenio_oaiserver/fetchers.py              10      0      2      0   100%
invenio_oaiserver/minters.py               20      0      2      0   100%
invenio_oaiserver/models.py                47      0      8      0   100%
invenio_oaiserver/percolator.py            70      0     36      0   100%
invenio_oaiserver/provider.py              15      0      4      0   100%
invenio_oaiserver/proxies.py                6      0      2      0   100%
invenio_oaiserver/query.py                113     79     52      1    30%
invenio_oaiserver/receivers.py             24      0      6      0   100%
invenio_oaiserver/response.py             426    148    158      1    62%
invenio_oaiserver/resumption_token.py      31      0      8      0   100%
invenio_oaiserver/schemas/__init__.py       1      0      0      0   100%
invenio_oaiserver/tasks.py                 28      0      6      0   100%
invenio_oaiserver/utils.py                121      0     42      0   100%
invenio_oaiserver/verbs.py                 75      0     48      0   100%
invenio_oaiserver/version.py                3      0      0      0   100%
invenio_oaiserver/views/__init__.py         1      0      0      0   100%
invenio_oaiserver/views/server.py          54      0     22      0   100%
-------------------------------------------------------------------------
TOTAL                                    1192    227    432      2    79%

=========================== short test summary info ============================
FAILED tests/test_query.py::test_get_records - elasticsearch.exceptions.Autho...
FAILED tests/test_response.py::test_getrecord - AssertionError: assert [] == ...
FAILED tests/test_response.py::test_check_correct_system_props_mapping - Attr...
FAILED tests/test_response.py::test_combine_record_file_urls - AttributeError...
ERROR tests/test_query.py::test_get_records - elasticsearch.exceptions.Author...
ERROR tests/test_response.py::test_listidentifiers - elasticsearch.exceptions...
ERROR tests/test_response.py::test_listrecords - elasticsearch.exceptions.Aut...
ERROR tests/test_response.py::test_issue34851_listrecords - elasticsearch.exc...
ERROR tests/test_response.py::test_issue34851_listidentifiers - elasticsearch...
ERROR tests/test_verbs.py::test_getrecord_fail - elasticsearch.exceptions.Aut...
ERROR tests/test_verbs.py::test_listmetadataformats_record - elasticsearch.ex...
= 4 failed, 78 passed, 3 skipped, 10827 warnings, 7 errors in 179.16s (0:02:59) =
ERROR: InvocationError for command /code/modules/invenio-oaiserver/.tox/c1/bin/pytest --cov=invenio_oaiserver tests -v -vv -s --cov-branch --cov-report=term --basetemp=/code/modules/invenio-oaiserver/.tox/c1/tmp (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   c1: commands failed
